tests/rados/pool_tests.py

```diff
@@ -400,3 +400,43 @@ def run(ceph_cluster, **kw):
             log.info(f"Completed the test of pg_min_num on pool: {entry['pool_name']} ")
         log.info("pg_min_num tests completed")
         return 0
+
+    if config.get("verify_pg_num_limit"):
+        """
+        This test is to verify the pg number limit for a pools.
+        Script cover the following steps-
+           1. Creating a pool with default pg number
+           2. Set the pg number more than the limit that is more than 128
+           3. Cheking that pg number getting reduced or not
+        """
+
+        pool_name = config.get("verify_pg_num_limit")["pool_name"]
+        rados_obj.create_pool(pool_name)
+
+        # set the pg number more than 128
+        cmd = f"ceph osd pool set {pool_name} pg_num 256"
+        rados_obj.run_ceph_command(cmd=cmd)
+        time.sleep(5)
+
+        # Checking the pg number is getting below 128
```

Comment:
Is the feature to work in accordance with PG_autoscaler? Or the count would reduce even with pg_autoscaler set to false?

Reply:
pg_autoscale should be on, default autoscale is on.so no need to change the scale.
 [ceph: root@ceph-test-rados-lu54di-node1-installer /]# ceph osd pool autoscale-status
POOL                     SIZE  TARGET SIZE  RATE  RAW CAPACITY   RATIO  TARGET RATIO  EFFECTIVE RATIO  BIAS  PG_NUM  NEW PG_NUM  AUTOSCALE  BULK   
device_health_metrics  13898                 3.0        324.9G  0.0000                                  1.0       1              on         False  
cephfs.cephfs.meta     47956                 3.0        324.9G  0.0000                                  4.0      16              on         False  
cephfs.cephfs.data         0                 3.0        324.9G  0.0000                                  1.0      32              on         False  
.rgw.root               1614                 3.0        324.9G  0.0000                                  1.0      32              on         False  
default.rgw.log         4114                 3.0        324.9G  0.0000                                  1.0      32              on         False  
default.rgw.control        0                 3.0        324.9G  0.0000                                  1.0      32              on         False  
default.rgw.meta         505                 3.0        324.9G  0.0000                                  4.0       8              on         False  
test_pool              17335M                3.0        324.9G  0.1563                                  1.0      64              on         False  
[ceph: root@ceph-test-rados-lu54di-node1-installer /]# client_loop: send disconnect: Broken pipe